source/Kepler-System/SystemModule.class.st

```diff
@@ -0,0 +1,91 @@
+"
+I'm a system module.
+I will install one or more subsystems, and perform all the neccessary post installation setup to make it work.
+"
+Class {
+	#name : #SystemModule,
+	#superclass : #Object,
+	#category : #'Kepler-System'
+}
+
+{ #category : #private }
+SystemModule >> >> anInterfaceKey [
+
+	^ self rootSystem >> anInterfaceKey
+]
+
+{ #category : #actions }
+SystemModule >> install [
+
+	self >> #InstalledModuleRegistrationSystem
+		ifNotRegisteredAsInstalled: self
+		do: [ 
+			self
+				installSystems;
+				postInstall ]
+]
+
+{ #category : #private }
+SystemModule >> installSystems [
+
+	"The modules can provide reflectively the sytemts ton isntall by implementing messages matching: register*SystemForInstallationIn:
+	Returning the system instance (not started)"
```

Comment:
Change the first part of the comment to:
Modules can reflectively provide the systems to install by implementing messages matching: register*SystemForInstallationIn: